src/demoserver.cpp

```diff
@@ -0,0 +1,294 @@
+        AOPacket ao_packet(packet);
+        handle_packet(ao_packet);
+    }
+}
+
+void DemoServer::handle_packet(AOPacket packet)
+{
+    packet.net_decode();
+
+    // This code is literally a barebones AO server
+    // It is wise to do it this way, because I can
+    // avoid touching any of this disgusting shit
+    // related to hardcoding this stuff in.
+
+    // Also, at some point, I will make akashit
+    // into a shared library.
+
+    QString header = packet.get_header();
+    QStringList contents = packet.get_contents();
+
+    if (header == "HI") {
+        client_sock->write("ID#0#DEMOINTERNAL#0#%");
+    }
+    else if (header == "ID") {
+        QStringList feature_list = {
+                "noencryption", "yellowtext",       "prezoom",
+                "flipping",     "customobjections", "fastloading",
+                "deskmod",      "evidence",         "cccc_ic_support",
+                "arup",         "casing_alerts",    "modcall_reason",
+                "looping_sfx",  "additive",         "effects",
+                "y_offset",     "expanded_desk_mods"};
+        client_sock->write("PN#0#1#%");
+        client_sock->write("FL#");
+        client_sock->write(feature_list.join('#').toUtf8());
+        client_sock->write("#%");
+    }
+    else if (header == "askchaa") {
+        client_sock->write("SI#");
+        client_sock->write(QString::number(num_chars).toUtf8());
+        client_sock->write("#0#1#%");
+    }
+    else if (header == "RC") {
+        client_sock->write(sc_packet.toUtf8());
+    }
+    else if (header == "RM") {
+        client_sock->write("SM#%");
+    }
+    else if (header == "RD") {
+        client_sock->write("DONE#%");
+    }
+    else if (header == "CC") {
+        client_sock->write("PV#0#CID#-1#%");
+    }
+    else if (header == "CT") {
+        if (contents[1].startsWith("/load"))
+        {
+          QString path = QFileDialog::getOpenFileName(nullptr, tr("Load Demo"), "logs/", tr("Demo Files (*.demo)"));
+          if (path.isEmpty())
+            return;
+          load_demo(path);
+          client_sock->write("CT#DEMO#Demo file loaded. Send /play or > in OOC to begin playback.#1#%");
+        }
+        else if (contents[1].startsWith("/play") || contents[1] == ">")
+        {
+          if (timer->interval() != 0 && !timer->isActive())
+          {
+            timer->start();
+            client_sock->write("CT#DEMO#Resuming playback.#1#%");
+          }
+          else
+          {
+            if (demo_data.isEmpty() && p_path != "")
+              load_demo(p_path);
+            playback();
+          }
+        }
+        else if (contents[1].startsWith("/pause") || contents[1] == "|")
+        {
+          int timeleft = timer->remainingTime();
+          timer->stop();
+          timer->setInterval(timeleft);
+          client_sock->write("CT#DEMO#Pausing playback.#1#%");
+        }
+        else if (contents[1].startsWith("/max_wait"))
```

Comment:
Literally not documented. At least put it in some kind of help function or at the intro message.

Reply:
You do it :^)

Reply:
done